### ISSUE TITLE
Networking V2: cannot unset dns_nameservers for the subnet

### ISSUE THREAD

**Issue by kayrus** [terraform-providers/terraform-provider-openstack#728]
### Affected Resource(s)
Please list the resources as a list, for example:
- openstack_networking_subnet_v2

Try to set dns_nameservers during the create action, then unset dns_nameservers and see, that they're not unset.

in my local environment this did the trick:

```diff
diff --git a/openstack/resource_openstack_networking_subnet_v2.go b/openstack/resource_openstack_networking_subnet_v2.go                                         
index dd46f63..da9bfbe 100644                                                                                                                                    
--- a/openstack/resource_openstack_networking_subnet_v2.go                                                                                                       
+++ b/openstack/resource_openstack_networking_subnet_v2.go
@@ -365,7 +365,8 @@ func resourceNetworkingSubnetV2Update(d *schema.ResourceData, meta interface{})
                        return err
                }
                hasChange = true
-               updateOpts.DNSNameservers = resourceSubnetDNSNameserversV2(d)
+               dnsNameservers := resourceSubnetDNSNameserversV2(d)
+               updateOpts.DNSNameservers = &dnsNameservers
        }
 
        if d.HasChange("host_routes") {
diff --git a/vendor/github.com/gophercloud/gophercloud/openstack/networking/v2/subnets/requests.go b/vendor/github.com/gophercloud/gophercloud/openstack/networking/v2/subnets/requests.go
index d2c4a29..b56d3cb 100644
--- a/vendor/github.com/gophercloud/gophercloud/openstack/networking/v2/subnets/requests.go
+++ b/vendor/github.com/gophercloud/gophercloud/openstack/networking/v2/subnets/requests.go
@@ -188,7 +188,7 @@ type UpdateOpts struct {
        GatewayIP *string `json:"gateway_ip,omitempty"`
 
        // DNSNameservers are the nameservers to be set via DHCP.
-       DNSNameservers []string `json:"dns_nameservers,omitempty"`
+       DNSNameservers *[]string `json:"dns_nameservers,omitempty"`
 
        // HostRoutes are any static host routes to be set via DHCP.
        HostRoutes *[]HostRoute `json:"host_routes,omitempty"`
```

**Comment by jtopjian** [comment]
```diff
-       DNSNameservers []string `json:"dns_nameservers,omitempty"`
+       DNSNameservers *[]string `json:"dns_nameservers,omitempty"`
```

Good catch

**Comment by kayrus** [comment]
should be fixed by #745

**Comment by jtopjian** [comment]
#745 did fix this. I will follow-up with an acceptance test.

**Comment by jtopjian** [comment]
Fixed in #745
